Remotion/Validation/UnitTests/RoleCollectors/AddingObjectValidationRuleCollector1Test.cs

```diff
@@ -46,54 +50,159 @@ public void SetUp ()
+      validationMessageFactoryStub
+          .Setup(e => e.CreateValidationMessageForObjectValidator(It.IsAny<IObjectValidator>(), _validatedType))
+          .Returns(new InvariantValidationMessage("expectedMessage"));
+
+      Func<ObjectValidationRuleInitializationParameters, IObjectValidator> validatorFactory = param => new FakeObjectValidator(param.ValidationMessage);
+
+      _addingObjectValidationRuleCollector.RegisterValidator(validatorFactory);
+      var result = ((IAddingObjectValidationRuleCollector)_addingObjectValidationRuleCollector).CreateValidationRule(validationMessageFactoryStub.Object);
+
+      var fakeObjectValidators = ((ObjectValidationRule<Customer>)result).Validators.Cast<FakeObjectValidator>().ToArray();
+      Assert.That(fakeObjectValidators.Length, Is.EqualTo(1));
+      Assert.That(fakeObjectValidators[0].ValidationMessage.ToString(), Is.EqualTo("expectedMessage"));
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void CreateValidationRule_WhenCalledTwice_InitializesDeferredInitializationValidationMessagesOnlyForNewlyRegisteredValidators ()
     {
+      var validationMessageFactoryStub1 = new Mock<IValidationMessageFactory>();
+      validationMessageFactoryStub1
+          .Setup(e => e.CreateValidationMessageForObjectValidator(It.IsAny<IObjectValidator>(), _validatedType))
+          .Returns(new InvariantValidationMessage("expectedMessage1"));
+
+      var validationMessageFactoryStub2 = new Mock<IValidationMessageFactory>();
+      validationMessageFactoryStub2
+          .Setup(e => e.CreateValidationMessageForObjectValidator(It.IsAny<IObjectValidator>(), _validatedType))
+          .Returns(new InvariantValidationMessage("expectedMessage2"));
+
+      Func<ObjectValidationRuleInitializationParameters, IObjectValidator> validatorFactory = param => new FakeObjectValidator(param.ValidationMessage);
+
+      _addingObjectValidationRuleCollector.RegisterValidator(validatorFactory);
+      var result1 = ((IAddingObjectValidationRuleCollector)_addingObjectValidationRuleCollector).CreateValidationRule(validationMessageFactoryStub1.Object);
+
+      var fakeObjectValidators1 = ((ObjectValidationRule<Customer>)result1).Validators.Cast<FakeObjectValidator>().ToArray();
+      Assert.That(fakeObjectValidators1.Length, Is.EqualTo(1));
+      Assert.That(fakeObjectValidators1[0].ValidationMessage.ToString(), Is.EqualTo("expectedMessage1"));
+
+      _addingObjectValidationRuleCollector.RegisterValidator(validatorFactory);
+      var result2 = ((IAddingObjectValidationRuleCollector)_addingObjectValidationRuleCollector).CreateValidationRule(validationMessageFactoryStub2.Object);
+
+      var fakeObjectValidators2 = ((ObjectValidationRule<Customer>)result2).Validators.Cast<FakeObjectValidator>().ToArray();
+      Assert.That(fakeObjectValidators2.Length, Is.EqualTo(2));
+      Assert.That(fakeObjectValidators2[0].ValidationMessage.ToString(), Is.EqualTo("expectedMessage1"));
```

Comment:
We may want to check for the whole object at once instead of checking each message separately:
```suggestion
      Assert.That(fakeObjectValidators2.Select(f => f.ValidationMessage.ToString()), Is.EquivalentTo(expectedMessages));
```
Not sure if you want to actually do it here, but I got feedback about checking things separately in multiple asserts when it would be more readable to have one assert with an expected list of objects.

Reply:
Since I have it already this way in `AddingPropertyValidationRuleCollector1Test.cs` (merged), I think I would leave it as it is, because the tests should follow a similar logic